### ISSUE TITLE
bun resolve to incorrect `graphql` package version when install `postgraphile`

### ISSUE THREAD

**Issue by olragon** [oven-sh/bun#3857]
### What version of Bun is running?

0.7.1

### What platform is your computer?

Darwin 21.6.0 x86_64 i386

### What steps can reproduce the bug?

```bash
mkdir bun-postgraphile
cd bun-postgraphile
bun init
bun add postgraphile -y
```

### What is the expected behavior?

Bun should resolve and install graphql@`15.8.0`

### What do you see instead?

`graphql@16.0.0-experimental-stream-defer.5` installed

`cat yarn.lock`

```
"graphql@>=0.9 <0.14 || ^14.0.2 || ^15.4.0":
  resolved "https://registry.npmjs.org/graphql/-/graphql-15.8.0.tgz"
graphql@>=0.10.0, "graphql@>=0.11 <=16", "graphql@>=0.9 <0.14 || ^14.0.2 || ^15.4.0", "graphql@>=0.9 <0.14 || ^14.0.2 || ^15.4.0 || ^16.3.0", "graphql@^0.6.0 || ^0.7.0 || ^0.8.0-b || ^0.9.0 || ^0.10.0 || ^0.11.0 || ^0.12.0 || ^0.13.0 || ^14.0.2 || ^15.0.0":
  resolved "https://registry.npmjs.org/graphql/-/graphql-16.0.0-experimental-stream-defer.5.tgz"
```

**Comment by Electroid** [comment]
Thanks for reporting, looks like Bun has a bug where it is not resolving the "latest" version correctly.

**Comment by dylan-conway** [comment]
This is fixed in bun `v1.0.5` and newer versions of bun
